src/Castle.ts

```diff
@@ -0,0 +1,218 @@
+import fetch from 'node-fetch';
+import { reduce } from 'lodash';
+import { IncomingHttpHeaders } from 'http2';
+import AbortController from 'abort-controller';
+
+const defaultApiUrl = 'https://api.castle.io';
+
+type TrackParameters = {
+  event: string;
+  user_id: string;
+  user_traits?: object;
+  created_at?: string;
+  context: {
```

Comment:
ruby sdks context defaults eg https://github.com/castle/castle-ruby/blob/master/lib/castle/context/default.rb

Reply:
Adding library, will double check with @wallin about the other ones (active and origin)

Reply:
The last two I mentioned are deprecated, so not adding. There's also checking of language headers in that context/default.rb file, but I was told to hold off on that, later version.